sql/core/src/test/scala/org/apache/spark/sql/CollationSQLExpressionsSuite.scala

```diff
@@ -2295,6 +2295,827 @@ class CollationSQLExpressionsSuite
+    val query = s"SELECT array_except(array('a', 'b', 'c'), array('c', 'd', 'e'));"
+    withSQLConf(SqlApiConf.DEFAULT_COLLATION -> collation) {
+      checkAnswer(
+        sql(query),
+        Seq(
+          Row(Seq("a", "b"))
+        )
+      )
+      // check result row data type
+      val dataType = ArrayType(StringType(collation), false)
+      assert(sql(query).schema.head.dataType == dataType)
+    }
+  }
+
+  test("array_union supports collation") {
+    val collation = "UNICODE"
+    val query = s"SELECT array_union(array('a', 'b', 'c'), array('a', 'c', 'd'));"
+    withSQLConf(SqlApiConf.DEFAULT_COLLATION -> collation) {
+      checkAnswer(
+        sql(query),
+        Seq(
+          Row(Seq("a", "b", "c", "d"))
+        )
+      )
+      // check result row data type
+      val dataType = ArrayType(StringType(collation), false)
+      assert(sql(query).schema.head.dataType == dataType)
+    }
+  }
+
+  test("array_compact supports collation") {
+    val collation = "UNICODE"
+    val query = s"SELECT array_compact(array('a', 'b', null, 'c'));"
+    withSQLConf(SqlApiConf.DEFAULT_COLLATION -> collation) {
+      checkAnswer(
+        sql(query),
+        Seq(
+          Row(Seq("a", "b", "c"))
+        )
+      )
+      // check result row data type
+      val dataType = ArrayType(StringType(collation), true)
+      assert(sql(query).schema.head.dataType == dataType)
+    }
+  }
+
+  test("arrays_zip supports collation") {
+    val collation = "UNICODE"
+    val query = s"SELECT arrays_zip(array('a', 'b', 'c'), array(1, 2, 3));"
+    withSQLConf(SqlApiConf.DEFAULT_COLLATION -> collation) {
+      checkAnswer(
+        sql(query),
+        Seq(
+          Row(Seq(Row("a", 1), Row("b", 2), Row("c", 3)))
+        )
+      )
+      // check result row data type
+      val dataType = ArrayType(StructType(
+        StructField("0", StringType(collation), true) ::
+          StructField("1", IntegerType, true) :: Nil
+      ), false)
+      assert(sql(query).schema.head.dataType == dataType)
+    }
+  }
+
+  test("array_min supports collation") {
+    val collation = "UNICODE"
+    val query = s"SELECT array_min(array('a', 'b', null, 'c'));"
+    withSQLConf(SqlApiConf.DEFAULT_COLLATION -> collation) {
+      checkAnswer(
+        sql(query),
+        Seq(
+          Row("a")
+        )
+      )
+      // check result row data type
+      val dataType = StringType(collation)
+      assert(sql(query).schema.head.dataType == dataType)
+    }
+  }
+
+  test("array_max supports collation") {
+    val collation = "UNICODE"
+    val query = s"SELECT array_max(array('a', 'b', null, 'c'));"
+    withSQLConf(SqlApiConf.DEFAULT_COLLATION -> collation) {
+      checkAnswer(
+        sql(query),
+        Seq(
+          Row("c")
+        )
+      )
+      // check result row data type
+      val dataType = StringType(collation)
+      assert(sql(query).schema.head.dataType == dataType)
+    }
+  }
+
+  test("array_append supports collation") {
```

Comment:
Appending a string with different collation fails (default collation is `UTF8_BINARY`):
```sql
SELECT array_append(array('b', 'd', 'c', 'a' COLLATE UTF8_LCASE), 'e')
```
while, `array_contains` passes:
```sql
SELECT array_contains(array('b', 'd', 'c', 'a' COLLATE UTF8_LCASE), 'e')
```
I'm a bit confused how implicit casting works here? @mihailom-db can you provide some insights?

Reply:
iiuc, arrays in spark can only hold elements of a single data type (i.e. dataType must be uniform within an ArrayType)

with this in mind - we should probably look into why we are allowed to do `array('a', 'b' COLLATE UTF8_LCASE)` in the first place; @mihailom-db and @stevomitric let's discuss offline and follow up asap

Reply:
update: consulted with @mihailom-db to conclude that this was missed by accident, so we will fix in a follow-up